### DISCUSSION TITLE
Styling text with multicharacter emojis in it

### DISCUSSION THREAD

**Discussion by bbernhard** [AsamK/signal-cli#1338] [Q&A] [answered]
Hi,

I am currently playing around with text formatting (`--text-style`) in signal-cli. Everything works fine if I format plain text, but if the text contains emojis, I am seeing some strange behavior (or at least I can't explain what's going on there).

e.g if I execute the following command

`echo '👋abcdefg' | signal-cli  --config /home/.local/share/signal-cli -a <+number> send --message-from-stdin <+recipient> --text-style 1:7:BOLD`

the text that I receive looks like this: 👋👋**abcdef**g

If I however change the formatting to `--text-style 2:8:BOLD` everything is fine again and the text that I receive looks like this: 👋**abcdefg**

Does anyone know, why the emoji basically shifts everything by one? The 👋 emoji is just a single UTF-8 character, right? So I would assume that `--text-style 1:7:BOLD` should be correct here? Or am I missing something here?

edit: my `LANG` environment variable is set to UTF-8.

Thanks!

**Comment by AsamK** [comment, marked as answer]
Java (and Android) use UTF-16 encoding, so they probably use this when counting chars. An emoji consists of two UTF-16 chars.
signal-cli just passes those numbers through and doesn't interpret them.

**Reply by bbernhard** [reply]
aaah, got it. Wasn't aware that Java uses UTF-16 encoding. Many thanks!
